commands/miscellaneous/getusersinserver.js

```diff
@@ -0,0 +1,76 @@
+const Discord = require('discord.js');
+const { SlashCommandBuilder } = require('@discordjs/builders');
+const { BATId, BALId, BAGId } = require('./config.json');
+
+const ALL_GUILD_IDS = [BATId, BALId, BAGId];
+
+// TODO: needs permission 'MANAGE_ROLES'
+module.exports = {
+  data: new SlashCommandBuilder()
+      .setName('getusersinserver')
+      .setDescription('Gets a list of user ids that were mentioned in a message and checks if they are in the satellite servers.')
+      .addStringOption(option =>
+          option.setName('message_ids')
+              .setDescription('The message id(s) you would like to get the user ids from')
+              .setRequired(true))
+      .addChannelOption(option =>
+          option.setName('channel')
+              .setDescription('The channel the message(s) are in')
+              .setRequired(true)),
+  async execute(interaction) {
+    const options = interaction.options;
+    const messageIds = options.getString('message_ids');
+    const channel = options.getChannel('channel');
+
+    const allUserIdsPerGuild = {};
```

Comment:
i moved the grab of members out of the `getUsersInGuild` so it isn't run on every messageId and we only grab this once and then check against the ids later